notifications/utils.py

```diff
@@ -46,3 +52,21 @@ def mark_as_sent_or_canceled(notification):
             log.exception("EmailNotification rolled back to pending: %s", notification.id)
             notification.state = EmailNotification.STATE_PENDING
             notification.save()
+
+
+@contextmanager
+def praw_error_to_cancelled():
+    """Raises CancelNotificationErrors from certain praw errors, otherwise raises the original error"""
+    try:
+        yield
+    except (Forbidden, NotFound, Redirect) as exc:
+        raise CancelNotificationError() from exc
+    except APIException as exc:
+        if exc.error_type in ('SUBREDDIT_NOTALLOWED', 'SUBREDDIT_NOEXIST', 'DELETED_COMMENT'):
+            raise CancelNotificationError() from exc
+        raise
```

Comment:
Can you add a test for this case, where the exception doesn't match any of the other cases?